doc/guides/maintaining-the-build-files.md

```diff
@@ -0,0 +1,386 @@
+    --shared-openssl-includes=SHARED_OPENSSL_INCLUDES
+                        directory containing OpenSSL header files
+    --shared-openssl-libname=SHARED_OPENSSL_LIBNAME
+                        alternative lib name to link to [default: crypto,ssl]
+    --shared-openssl-libpath=SHARED_OPENSSL_LIBPATH
+                        a directory to search for the shared OpenSSL DLLs
+    --shared-zlib       link to a shared zlib DLL instead of static linking
+    --shared-zlib-includes=SHARED_ZLIB_INCLUDES
+                        directory containing zlib header files
+    --shared-zlib-libname=SHARED_ZLIB_LIBNAME
+                        alternative lib name to link to [default: z]
+    --shared-zlib-libpath=SHARED_ZLIB_LIBPATH
+                        a directory to search for the shared zlib DLL
+    --shared-cares      link to a shared cares DLL instead of static linking
+    --shared-cares-includes=SHARED_LIBCARES_INCLUDES
+                        directory containing cares header files
+    --shared-cares-libname=SHARED_LIBCARES_LIBNAME
+                        alternative lib name to link to [default: cares]
+    --shared-cares-libpath=SHARED_LIBCARES_LIBPATH
+                        a directory to search for the shared cares DLL
+
+  Internationalization:
+    Flags that lets you enable i18n features in Node.js as well as which
+    library you want to build against.
+
+    --with-intl=WITH_INTL
+                        Intl mode (valid choices: none, small-icu, full-icu,
+                        system-icu) [default: small-icu]
+    --without-intl      Disable Intl, same as --with-intl=none (disables
+                        inspector)
+    --with-icu-path=WITH_ICU_PATH
+                        Path to icu.gyp (ICU i18n, Chromium version only.)
+    --with-icu-locales=WITH_ICU_LOCALES
+                        Comma-separated list of locales for "small-icu".
+                        "root" is assumed. [default: root,en]
+    --with-icu-source=WITH_ICU_SOURCE
+                        Intl mode: optional local path to icu/ dir, or
+                        path/URL of icu source archive.
+    --download=DOWNLOAD_LIST
+                        Select which packages may be auto-downloaded. valid
+                        values are: none, all, icu. (default is "none").
+    --download-path=DOWNLOAD_PATH
+                        Download directory [default: ./deps]
+
+  HTTP2:
+    Flags that allows you to control HTTP2 features in Node.js
+
+    --debug-http2       build with http2 debug statements on (default is
+                        false)
+    --debug-nghttp2     build nghttp2 with DEBUGBUILD (default is false)
+```
+
+## Makefile
+
+This file is not generated and mantained by humans. Note that this is
+not usually run on Windows, where [vcbuild.bat](#vcbuild.bat) is used instead.
+
+### Options
+
+- `-j <n>`: number of threads used to build the binary or run the parallel 
+  tests
```

Comment:
Parallel tests are run in parallel by default (by virtue of `-J` specified to the test runner in Makefile), even if the `make` command is `-j1`.

Reply:
There's an awkward catch to this that's not documented anywhere that probably should be since I'm finding Jenkins jobs that don't do this properly: a lot of the -ci targets use this `PARALLEL_ARGS` variable to insert `-j X` for `tools/test.py` (`-J` isn't used in any of the -ci targets). Unfortunately this variable is only set if `$JOBS` exists, and this isn't automatically set just because you supply `-j X` to `make`. In fact, it's kind of difficult to determine the `-j` argument inside a Makefile in a cross-platform way (but we should probably try, there's [an interesting stackoverflow suggestion here](https://stackoverflow.com/questions/5303553/gnu-make-extracting-argument-to-j-within-makefile)). So you have to give `JOBS=X` when running the -ci targets.

`-J` is done automatically for tests on non-ci targets so it takes up "as many cores as you have" for tests as @TimothyGu said.

Reply:
@TimothyGu @rvagg Thanks for the correction, I'll add a note in the Makefile about this, and change this part to just "number of threads used to build the binary"